pinot-core/src/main/java/org/apache/pinot/core/query/pruner/AbstractSegmentPruner.java

```diff
@@ -87,26 +89,17 @@ protected boolean pruneNonLeaf(@Nonnull FilterQueryTree filterQueryTree,
    * @param input Input String for which to get the value
    * @param dataType Data type to construct from the String.
    * @return Comparable value of specified data type built from the input String.
-   * @note It is assumed that the 'input' here is a value taken from the query, so this method
-   * should not be used to for other internal purposes.
+   * @apiNote It is assumed that the 'input' here is a value taken from the query, so this method should not be used for
+   * other internal purposes.
    */
   protected static Comparable getValue(@Nonnull String input, @Nonnull FieldSpec.DataType dataType) {
     try {
-      switch (dataType) {
-        case INT:
-          return Integer.valueOf(input);
-        case LONG:
-          return Long.valueOf(input);
-        case FLOAT:
-          return Float.valueOf(input);
-        case DOUBLE:
-          return Double.valueOf(input);
-        case STRING:
-          return input;
-        default:
-          throw new IllegalStateException("Unsupported data type: " + dataType);
+      if (dataType != FieldSpec.DataType.BYTES) {
```

Comment:
Is there a way we can eliminate this special check for BYTES?

Reply:
If we only keep ByteArray as the value inside Pinot, we can achieve that. Right now we have mixed of byte[] and ByteArray, thus hard to uniform all behaviors. We might want to change the code to use only ByteArray internal in a separate PR.

Reply:
Internally we mix the use of byte[] and ByteArray, which causes this special handling. IMO we should uniform the behavior to always use ByteArray (at least for segment generation, and filters where comparison applies). That will involve much bigger changes, which can be submitted in a separate PR if we decide to do so.